### DISCUSSION TITLE
Using 3 gas sensors, but only 2 return values

### DISCUSSION THREAD

**Discussion by jankornblum** [HomeSpan/HomeSpan#483] [Q&A] [answered]
My device is using 3 gas sensors (MQ-135 for CO2, MQ-9 for CO, MQ-4 for methane) using [MQSensorLib](https://github.com/miguel5612/MQSensorsLib) . Without using homespan.poll() everything works fine and each sensors value can be read in main loop() function.

But as soon as homespan.poll() gets called inside main loop() function, the third sensors value cannot be read any longer. Neither in main loop() function nor inside the homespan accessory service loop() function.

What might be the reason?

**Comment by HomeSpan** [comment, marked as answer]
Does the order of the sensors matter.  In other words, does everything work fine with two specific sensors, but one particular one does not work.  Or is it the case that only the first two sensors (regardless of type) work and the third (regardless of type) does not work.  This will allow you to determine whether the issue is specific to the type of sensor.

**Reply by jankornblum** [reply]
@HomeSpan Thx a lot. Indeed the sensor itself works with homespan. The problem is rather GPIO related. The sensors don’t work with HomeSpan on certain GPIOs (25, 27 and others). But using the same GPIOs and sensors without HomeSpan instead works fine. So are there any „homespan only“ related GPIO requirements?

**Reply by HomeSpan** [reply]
There are no dedicated HomeSpan-specific GPIOs.  Things like the Control Button and Status LED are optional, and if needed can use any pin.  However, HomeSpan does use WiFi and certain pins cannot be used for *analog* input at the same time WiFi is being used (note *digital* input is fine).  My guess is your sensor may be using analog (ADC) input, which cannot be done on pins 25 or 27 while using WiFi.

See https://arduino.stackexchange.com/questions/84888/analog-read-not-working-while-using-wifi for a complete explanation, including which ADC pins you *can* use without a conflict.

**Reply by jankornblum** [reply]
@HomeSpan Yes it is ADC. That clarifies anything… Thanks a lot for this hint!
